### PR TITLE
CONSOLE-3055: Introduction of multi cluster menu filter capability

### PR DESCRIPTION
**Pull request by sg00dwin**
Convert existing cluster `Dropdown` to use PatternFly `MenuToggle` and `Menu` components, which includes filtering with search input option, along with built in capabilities to support grouping, isSelected and favoriting functionality. 

https://issues.redhat.com/browse/CONSOLE-3055

- [x] Add sort to prevent change of order
- [x] Add alphanumeric sorting similar to namespace menu (https://github.com/openshift/console/pull/12428)

cluster-menu-selector

### CODE REVIEW

**Review by TheRealJon — CHANGES_REQUESTED** [review]
A couple of adjustments, but it looks good so far! One recommendation I have would be to break out a new ClusterDropdown component. We have a lot of dropdown management logic in the NavHeader component that arguably doesn't belong here.

File: frontend/packages/console-app/src/components/nav/NavHeader.tsx [hunk 1]
```diff
@@ -185,48 +235,102 @@ const NavHeader: React.FC<NavHeaderProps> = ({ onPerspectiveSelected }) => {
...
-                    t('console-app~All Clusters')
-                  ) : (
-                    <>
-                      <ClusterIcon /> {activeCluster}
-                    </>
-                  )}
-                </Title>
-              </DropdownToggle>
+          <ClusterMenuToggle
+            disabled={false}
```
[the remark this answers is not in the document]
Comment (sg00dwin) [reply frontend/packages/console-app/src/components/nav/NavHeader.tsx:320]: If I remove this it give the following error `Property 'disabled' is missing in type`

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Some adjustments and a couple of concerns:
- "All Clusters" might not show a selected state when it is the currently selected item
- "All Clusters" item seems to be excluded from filtering

**Comment by jhadvig** [comment]
@sg00dwin we need to rebase this one.

File: frontend/packages/console-app/src/components/nav/ClusterMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          <MenuItem
+            translate="no"
+            key={cluster.key}
+            itemId={cluster.key}
+            isSelected={activeCluster === cluster.key}
+            onClick={(e) => {
+              e.preventDefault();
+              cluster.onClick();
+            }}
+          >
```
Comment (jcaianirh) [inline frontend/packages/console-app/src/components/nav/ClusterMenu.tsx:85]: data-test-id="cluster-dropdown-item"

add this for e2e tests to access the menu item. if you want a better name, then the test id used here should match the one in nav.ts...ie cluster-menu-item 

cc: @TheRealJon

File: frontend/packages/console-app/src/components/nav/ClusterMenuToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const menuToggle = (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={() => onToggle(!isOpen)}
+      isExpanded={isOpen}
+      variant="plainText"
+      isFullWidth
+      disabled={disabled}
+      className="co-cluster-selector"
+    >
```
Comment (jcaianirh) [inline frontend/packages/console-app/src/components/nav/ClusterMenuToggle.tsx:58]: data-test-id="cluster-dropdown-toggle"

same as previous data-test-id comment wrt id name value

**Review by jcaianirh — CHANGES_REQUESTED** [review]
Aside from the changes requested, theres a bug with this code as follows:
When viewing a workload in a namespace on the hub cluster, then change the cluster to a spoke (or vice versa) the namespace value becomes the string undefined causing errors on the page. The undefined string also appears in the url for the page. I haven't found why/where that code change is but I can try and help find it.

**Review by jcaianirh — APPROVED** [review]

**Comment by jcaianirh** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

Console Approver:
/assign @rhamilto

**Comment by spadgett** [comment]
@rhamilto FYI since you're working on the masthead changes.

**Comment by RickJWagner** [comment]
/label px-approved

Thank you for providing the screen capture in this issue.

**Comment by yapei** [comment]
1. it looks like the cluster filter menu boundary will be enlarged when no results match the filter criteria
no-clusters-found

**Comment by yapei** [comment]
tested and no medium above issue found, clusters are ordered correct in filter dropdown
/label qe-approved

**Pull request merged**